Source/SwiftyJSON.swift

```diff
@@ -1293,6 +1293,50 @@ public func <(lhs: JSON, rhs: JSON) -> Bool {
     }
 }
 
+// MARK: - Find
+extension JSON {
+    /**
+    Find an underlying `JSON` object using a depth-first search.
+     
+     - parameter predicate: A method that returns `true` for only a target `JSON` object
+     
+     - returns: The first `JSON` object to satisfy the predicate or `nil` if none does.
+    */
+    func find(@noescape predicate predicate: JSON -> Bool) -> JSON? {
+        if predicate(self) {
+            return self
+        }
+        else if let subJSON = (dictionary?.map { $0.1 } ?? array) {
```

Comment:
Haha you're right! I didn't even realize that when I was writing it. However, sometimes I like to add it in order to make it more apparent that only the code inside one of the conditionals can be run. I definitely agree, though.